### ISSUE TITLE
`amplify push` throws error "Maximum call stack size exceeded" tracing to `amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js`

### ISSUE THREAD

**Issue by ivadenis** [aws-amplify/amplify-cli#12244]
### How did you install the Amplify CLI?

yarn

### If applicable, what version of Node.js are you using?

14

### Amplify CLI Version

11

### What operating system are you using?

Mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

no

### Describe the bug

After upgrading CLI to 11.0.0 `amplify push` throws an error `DeploymentFault: Maximum call stack size exceeded`

stack trace from `amplify push --debug`

### Expected behavior

it should not throw error on amplify push

### Reproduction steps

1. amplify upgrade
2. amplify push

sorry, these are not really helpful...

### Project Identifier

b3d7bfaada2407aa41cf925f10add28d

### Log output

<details>

```
# Put your logs below this line



DeploymentFault: Maximum call stack size exceeded
    at Object.run (/snapshot/repo/build/node_modules/@aws-amplify/amplify-provider-awscloudformation/lib/push-resources.js:455:11)
    at async Promise.all (index 0)
    at async providersPush (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/extensions/amplify-helpers/push-resources.js:159:5)
    at async AmplifyToolkit.pushResources (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/extensions/amplify-helpers/push-resources.js:134:13)
    at async Object.executeAmplifyCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/index.js:189:9)
    at async executePluginModuleCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/execution-manager.js:135:5)
    at async executeCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/execution-manager.js:33:9)
    at async Object.run (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/index.js:117:5)

Maximum call stack size exceeded
RangeError: Maximum call stack size exceeded
    at Array.values (<anonymous>)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:79:55)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
    at YarnLockParser.dfs (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:88:30)
   
    ....

    at YarnLockParser.getDependentPackageMap (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/yarn-lock-parser.js:66:26)
    at AmplifyNodePkgDetector.detectAffectedDirectDependencies (/snapshot/repo/build/node_modules/amplify-cli-core/lib/amplify-node-pkg-detector/amplify-nodejs-detector.js:51:67)
    at getOverridesWarning (/snapshot/repo/build/node_modules/@aws-amplify/amplify-provider-awscloudformation/lib/print-cdk-migration-warning.js:93:104)
    at getMigrationMessage (/snapshot/repo/build/node_modules/@aws-amplify/amplify-provider-awscloudformation/lib/print-cdk-migration-warning.js:132:34)
    at printCdkMigrationWarning (/snapshot/repo/build/node_modules/@aws-amplify/amplify-provider-awscloudformation/lib/print-cdk-migration-warning.js:74:59)
    at async Object.run (/snapshot/repo/build/node_modules/@aws-amplify/amplify-provider-awscloudformation/lib/push-resources.js:236:5)
    at async Promise.all (index 0)
    at async providersPush (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/extensions/amplify-helpers/push-resources.js:159:5)
    at async AmplifyToolkit.pushResources (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/extensions/amplify-helpers/push-resources.js:134:13)
    at async Object.executeAmplifyCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/index.js:189:9)
    at async executePluginModuleCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/execution-manager.js:135:5)
    at async executeCommand (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/execution-manager.js:33:9)
    at async Object.run (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/index.js:117:5)


```

</details>

### Additional information

after downgrading to 10.8.0 no error is thrown

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by sobolk** [comment]
Fix will ship with 11.1. We're sorry for the inconvenience.

**Comment by josefaidt** [comment]
Closing as this has since been fixed and released with 11.0.3 🎉
